### PR TITLE
Add types

### PR DESCRIPTION
**Pull request by Roang-zero1**
Add types and tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,30 +19,40 @@
...
+  "files": [
+    "index.js",
+    "index.d.ts"
+  ],
+  "types": "index.d.ts",
   "dependencies": {
     "unist-util-is": "^4.0.0"
   },
   "devDependencies": {
+    "@types/tape": "^4.2.33",
```
Comment (wooorm) [inline package.json]: Where are these used?

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import {Node, Parent} from 'unist'
+import {Test} from 'unist-util-is/index'
```
Comment (wooorm) [inline index.d.ts]: Why is the `/index` needed?

File: package.json (hunk 1 above)
Comment (Roang-zero1) [reply package.json]: I use typescript to verify the JavaScript code.
In order to correctly verify the tests I had to add the types for the tape library.

File: index.d.ts (hunk 2 above)
Comment (Roang-zero1) [reply index.d.ts]: If I try to import the Test interface from `unist-util-is` without the `/index` dtslint can't find the interface and will error out.

Comment (ChristianMurphy) [reply index.d.ts]: @Roang-zero1 could you try removing it again?
Testing locally `dtslint` works with `import {Test} from 'unist-util-is'`

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import {Node, Parent} from 'unist'
```
Comment (ChristianMurphy) [inline index.d.ts]: typings should usually note the minimum supported typescript version.
In this case it depends on unist-util-is which needs at least `3.5`:

```suggestion
// TypeScript Version: 3.5

import {Node, Parent} from 'unist'
```

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "compilerOptions": {
+    "lib": ["es2015"],
+    "strict": true,
+    "skipLibCheck": true,
```
Comment (ChristianMurphy) [inline tsconfig.json]: The typing re-exports types from other libraries, lib check should be enabled.

```suggestion
```

File: index.d.ts (hunk 2 above)
Comment (Roang-zero1) [reply index.d.ts]: Tried again and it worked.

**Comment by Roang-zero1** [comment]
Reviews have been addressed.

**Comment by wooorm** [comment]
Released!

**Pull request merged**